apm-sniffer/apm-sdk-plugin/micronaut-plugins/micronaut-http-client-plugin/src/main/java/org/apache/skywalking/apm/plugin/micronaut/http/client/ClientConstructorInterceptPoint.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.apm.plugin.micronaut.http.client;
+
+import io.micronaut.http.client.LoadBalancer;
+import io.micronaut.http.client.loadbalance.FixedLoadBalancer;
+import io.micronaut.http.client.loadbalance.ServiceInstanceListRoundRobinLoadBalancer;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.EnhancedInstance;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.InstanceConstructorInterceptor;
+
+public class ClientConstructorInterceptPoint implements InstanceConstructorInterceptor {
+    //Hold the target address , it maybe serviceId or domain address
+    //Micronaut 3.6.x , we get the address from io.micronaut.http.client.netty.DefaultHttpClient.informationalServiceId
+    //Micronaut 3.2.x-3.5.x , we can only get it from LoadBalancer
+    @Override
+    public void onConstruct(EnhancedInstance enhancedInstance, Object[] objects) throws Throwable {
+        // Micronaut 3.6.x
+        if (objects.length == 17) {
+            if (objects[16] != null) {
```

Comment:
This is not recommended, you could use two instrumentations to target various versions. That is 100% no runtime impact.